src/test/java/com/hotels/hcommon/hive/metastore/iterator/PartitionIteratorTest.java

```diff
@@ -147,4 +156,51 @@ public void noPartitions() throws MetaException, TException {
     assertThat(iterator.hasNext(), is(false));
   }
 
+  @Test
+  public void reverseOrderWithBatching() throws MetaException, TException {
+    when(metastore
+        .getPartitionsByNames(DATABASE_NAME, TABLE_NAME,
+            Arrays.asList("year=2016/month=08/day=04", "year=2014/month=05/day=13", "year=2014/month=03/day=24")))
+                .thenReturn(Arrays.asList(p20140324, p20140513, p20160804));
+    when(metastore
+        .getPartitionsByNames(DATABASE_NAME, TABLE_NAME,
+            Arrays.asList("year=2014/month=03/day=05", "year=2013/month=10/day=11", "year=2013/month=10/day=09")))
+                .thenReturn(Arrays.asList(p20131009, p20131011, p20140305));
+    when(metastore
+        .getPartitionsByNames(DATABASE_NAME, TABLE_NAME,
+            Arrays.asList("year=2012/month=01/day=04", "year=2012/month=01/day=03", "year=2012/month=01/day=02")))
+                .thenReturn(Arrays.asList(p20120102, p20120103, p20120104));
+
+    when(metastore.getPartitionsByNames(DATABASE_NAME, TABLE_NAME, Arrays.asList("year=2012/month=01/day=01")))
+        .thenReturn(Arrays.asList(p20120101));
+
+    PartitionIterator iterator = new PartitionIterator(metastore, table, (short) 3, Ordering.REVERSE);
+    assertThat(iterator.hasNext(), is(true));
```

Comment:
I see that you are doing these multiple asserts in three tests. Can these asserts be refactored into a assertFunction with a List of expected Partitions as argument.

Reply:
I'm gonna leave that for now, we could probably do with a lot less partitions for testing tbh.